### PR TITLE
prowgen: generate stricter `branch:` stanzas for presubmits

### PR DESCRIPTION
**Pull request by petr-muller**
In the past, we considered any `$branch$suffix` a feature branch of
"base" CI branch (e.g. PRs to `master-next` would receive presubmits
from `master` config). We need to make this convention stricter to not
break on `4.10`, because the current convention means that
`release-4.10` is a feature branch for `release-4.1`.

All generated presubmits now have two items in their `branches:` stanza.
One item that matches exactly the branch (`^branch$`), just like
postsubmits have, and one item that matches branches with `$base-`
prefix. Practically this means that we enforce exactly the `-` separator
for a branch to be considered a feature branch.

### CODE REVIEW

**Comment by petr-muller** [comment]
/hold
Needs a o/release followup after merge

**Review by stevekuznetsov — COMMENTED** [review]
Can we get a unit test for this?

**Comment by petr-muller** [comment]
> Can we get a unit test for this?
@stevekuznetsov hm, do you have a specific one in mind? `GeneratePresubmitForTest` is already unit tested (`TestGeneratePresubmitForTest`) and I just needed to change its fixtures to reflect the changed expectations about its output. Do you think there's any specific case that we would need to test?

**Review by stevekuznetsov — APPROVED** [review]
I was thinking something at a lower level than the full fixture-based thing but whatever

/lgtm
/hold cancel

**Pull request merged**
